sdk/state/update.go

```diff
@@ -97,6 +99,10 @@ func (c *Channel) ConfirmPayment(p *Payment) (payment *Payment, fullySigned bool
 		return nil, fullySigned, errors.New(fmt.Sprintf("invalid payment iteration number, got: %s want: %s",
 			strconv.FormatInt(p.IterationNumber, 10), strconv.FormatInt(c.NextIterationNumber(), 10)))
 	}
+	if c.latestUnconfirmedPayment != nil && !reflect.DeepEqual(c.latestUnconfirmedPayment, p) {
```

Comment:
ya  good catch, they should have different signatures. Test was passing because storing the pointer and then passing it around. I changed it to pass around payment as a value, and to store as a value. Also changed the `CloseAgreement` to store as value to prevent a similar test passing there when we add it

https://github.com/stellar/experimental-payment-channels/pull/81/commits/68688f7161180713daa554e55f57e4db37e89dc1